app/javascript/listings/elements/categories.jsx

```diff
@@ -3,54 +3,52 @@ import { h, Component } from 'preact';
 
 class Categories extends Component {
   options = () => {
-    const { categoriesForSelect, category } = this.props
-    return categoriesForSelect.map(array => {
+    const { categoriesForSelect, category } = this.props;
+    return categoriesForSelect.map(([text, value]) => {
```

Comment:
This is the refactoring here. The rest below line 20 is just prettier kicking in.